File: LeetCode/[M] - 1657. Determine if Two Strings Are Close/solution.go
package main

import "sort"

func closeStrings(word1 string, word2 string) bool {
	seq1, seq2 := make([]int, 26), make([]int, 26)

	for _, ch := range word1 {
		seq1[int(ch)-int('a')] += 1
	}
	for _, ch := range word2 {
		seq2[int(ch)-int('a')] += 1
	}
	for i := 0; i < 26; i++ {
		if (seq1[i] == 0 && seq2[i] != 0) || (seq1[i] != 0 && seq2[i] == 0) {
			return false
		}
	}
	sort.Ints(seq1)
	sort.Ints(seq2)

	for i := 0; i < 26; i++ {
		if seq1[i] != seq2[i] {
			return false
		}
	}
	return true
}

File: LeetCode/[M] - 1657. Determine if Two Strings Are Close/solution.py
class Solution:
    def closeStrings(self, word1: str, word2: str) -> bool:
        if len(word1) != len(word2):
            return False
        if set(word1) != set(word2):
            return False
        wordCnt1 = {}
        for word in word1:
            wordCnt1[word] = wordCnt1.get(word, 0) + 1
        wordCnt2 = {}
        for word in word2:
            wordCnt2[word] = wordCnt2.get(word, 0) + 1
        if sorted(wordCnt1.values()) == sorted(wordCnt2.values()):
            return True
        return False
